napari/_qt/widgets/qt_keyboard_settings.py

```diff
@@ -581,7 +581,16 @@ def event(self, event):
         if event.type() == QEvent.Type.ShortcutOverride:
             self.keyPressEvent(event)
             return True
-        if event.type() in [QEvent.Type.KeyPress, QEvent.Type.Shortcut]:
+        if event.type() == QEvent.Type.Shortcut:
+            return True
+        if event.type() == QEvent.Type.KeyPress and event.key() in (
+            Qt.Key.Key_Delete,
+            Qt.Key.Key_Backspace,
+        ):
+            # Need to mark as handled `Delete` and `Backspace` `KeyPress` events
+            # in order to be able to use those keys to remove shortcuts.
+            # To check the logic to remove shortcuts you can check the
+            # `KeyPressEvent` method
```

Comment:
```suggestion
            # If there is alreaty shortcut set, There are two events emmited when
            # press `Delete` or `Backspace`. First `ShortcutOverride`, then `KeyPress`.
            # So we need to mark second event as handeld and to not process it.  
            # Without it, Backspace/Delete will be set as shortcut instead of just clean.
```

Maybe try to describe in this way.

Reply:
Thanks for checking in more detail what was happening! If I'm understading correctly, could we leave as comment something like:

```
            # If there is a shortcut set already, two events are being emitted when
            # pressing `Delete` or `Backspace`. First a `ShortcutOverride` event and
            # then a `KeyPress` event. We need to mark the second event (`KeyPress`)
            # as handled for those keys to not end up processing it multiple times.
            # Without that, pressing `Backspace` or `Delete` will set those keys as shortcuts
            # instead of just cleaning/removing the previous shortcut that was set.
```

Used your suggestion for the comment structure and fixed a couple of typos but overall tried to communicate the same info. Let me know if the comment in that way is what you are expecting :)

Reply:
Your comment is much better than mine. Please use it.